internal/controllers/vgmanager/filter/filter_test.go

```diff
@@ -23,14 +24,111 @@ type advancedFilterTestCase struct {
 	lvmExpect []lvm.PhysicalVolume
 }
 
+func TestNewDeviceInStaticVolumeGroup(t *testing.T) {
```

Comment:
How do we test the y-stream upgrade path where we go from the field not existing, to existing and not being set to a value?

Reply:
So not being set to a value is not an option due to 
```
	// +kubebuilder:default=RuntimeDynamic
	// +kubebuilder:validation:Required
```
This will force the object to automatically contain RuntimeDynamic if no value is set. Similarly we can assume that the field will be present even for clusters that didnt have it set originally.

That said, the y-stream path would be to do a regular regression run where the LVMCluster gets created with a version that does not have this field change, then updating to the new CRD version, and then running a regression test. There should be no change in behavior. On the next edit the storage version would automatically get the field written in during serialization. I think this is pretty standard K8s behavior.